### PR TITLE
feat(client-logs-format) - change logs into the new format

### PR DESCRIPTION
**Pull request by nevobonneh**

### CODE REVIEW

File: client/src/Utils/ControllerHooks/useExposuresSaving.ts [hunk 1]
```diff
@@ -46,6 +44,12 @@ const useExposuresSaving = (exposuresAndFlightsVariables: ExposureAndFlightsDeta
     const saveExposureAndFlightData = async () : Promise<void> => {
         let { exposures, wereFlights, wereConfirmedExposures, exposuresToDelete } = exposuresAndFlightsVariables.exposureAndFlightsData;
         let filteredExposures : (Exposure | DBExposure)[] = [];
+        const saveExposureAndFlightDataLogger = logger.setup({
+            workflow: 'Saving Exposures And Flights',
+            service: Service.CLIENT,
```
Comment (ItayBenMoshe) [inline client/src/Utils/ControllerHooks/useExposuresSaving.ts:49]: You don't need to pass it service. it does it automatically

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

File: client/src/Utils/ControllerHooks/useExposuresSaving.ts (hunk 1 above)
Comment (ItayBenMoshe) [reply client/src/Utils/ControllerHooks/useExposuresSaving.ts:49]: delete it from all the setups

Comment (nevobonneh) [reply client/src/Utils/ControllerHooks/useExposuresSaving.ts:49]: so why is he an optiontal parameter in setup?

Comment (NaamanTsur) [reply client/src/Utils/ControllerHooks/useExposuresSaving.ts:49]: It probably doesn't need to be there

Comment (nevobonneh) [reply client/src/Utils/ControllerHooks/useExposuresSaving.ts]: fixed

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts [hunk 2]
```diff
@@ -64,14 +69,7 @@ const useInvestigatedPersonInfo = (): InvestigatedPersonInfoOutcome => {
...
                 if (result.value) {
-                    logger.info({
-                        service: Service.CLIENT,
-                        severity: Severity.LOW,
-                        workflow: 'Update Investigation Status',
-                        step: `launching investigation status request`,
-                        user: userId,
-                        investigation: epidemiologyNumber
-                    });
+                    updateInvestigationStatusLogger.info(`launching investigation status request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts:72]: change to singel quote

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts [hunk 3]
```diff
@@ -81,23 +79,9 @@ const useInvestigatedPersonInfo = (): InvestigatedPersonInfoOutcome => {
...
                         }).then(() => {
-                            logger.info({
-                                service: Service.CLIENT,
-                                severity: Severity.LOW,
-                                workflow: 'Update Investigation Status',
-                                step: `update investigation status request was successful`,
-                                user: userId,
-                                investigation: epidemiologyNumber
-                            });
+                            updateInvestigationStatusLogger.info(`update investigation status request was successful`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts:82]: same as above

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 4]
```diff
@@ -53,130 +53,72 @@ const useClinicalDetails = (parameters: useClinicalDetailsIncome): useClinicalDe
...
-            severity: Severity.LOW,
+        const getSymptomsLogger = logger.setup({
             workflow: 'Fetching Symptoms',
-            step: `launching symptoms request`,
-            user: userId,
-            investigation: epidemiologyNumber
+            investigation: epidemiologyNumber,
+            user: userId
         });
+        getSymptomsLogger.info(`launching symptoms request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:61]: same

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 5]
```diff
@@ -53,130 +53,72 @@ const useClinicalDetails = (parameters: useClinicalDetailsIncome): useClinicalDe
...
-            severity: Severity.LOW,
+        const getBackgroundDiseasesLogger = logger.setup({
             workflow: 'Fetching Background Diseases',
-            step: `launching background diseases request`,
-            user: userId,
-            investigation: epidemiologyNumber
+            investigation: epidemiologyNumber,
+            user: userId
         });
+        getBackgroundDiseasesLogger.info(`launching background diseases request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:79]: same

File: client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts [hunk 6]
```diff
@@ -53,130 +53,72 @@ const useClinicalDetails = (parameters: useClinicalDetailsIncome): useClinicalDe
...
-            severity: Severity.LOW,
+        const getIsolationSourcesLogger = logger.setup({
             workflow: 'Fetching Isolation Sources',
-            step: `Start isolation sources request`,
-            user: userId,
-            investigation: epidemiologyNumber
+            investigation: epidemiologyNumber,
+            user: userId
         });
+        getIsolationSourcesLogger.info(`Start isolation sources request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts:97]: same

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposureForm/ExposureForm.tsx [hunk 7]
```diff
@@ -79,32 +79,22 @@ const ExposureForm = (props: any) => {
   }
 
   useEffect(() => {
+    const confirmedExposuresLogger = logger.setup({
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposureForm/ExposureForm.tsx:82]: create confirmedExposuresLogger inside the else (you use this logger only there)

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx [hunk 8]
```diff
@@ -73,25 +73,23 @@ const ExposuresAndFlights: React.FC<Props> = ({ id }: Props): JSX.Element => {
...
+      user: userId
     });
+
+    const getCoronaTestDateLogger = logger.setup({
+      workflow: 'Getting Corona Test Date',
+      investigation: investigationId,
+      user: userId
+    });
+
+    fetchExposuresAndFlightsLogger.info(`launching exposures and flights request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx:88]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx [hunk 9]
```diff
@@ -107,56 +105,23 @@ const ExposuresAndFlights: React.FC<Props> = ({ id }: Props): JSX.Element => {
...
           }).catch((error) => {
-            logger.error({
-              service: Service.CLIENT,
-              severity: Severity.HIGH,
-              workflow: 'Fetching investigated patient resorts data',
-              step: `failed to get resorts response due to ` + error,
-              user: userId,
-              investigation: investigationId
-          });
+            getCoronaTestDateLogger.error(`failed to get resorts response due to ` + error,Severity.HIGH)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx:108]: use the $ syntax

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx [hunk 10]
```diff
@@ -107,56 +105,23 @@ const ExposuresAndFlights: React.FC<Props> = ({ id }: Props): JSX.Element => {
...
       .then(() => {
-        logger.info({
-          service: Service.CLIENT,
-          severity: Severity.LOW,
-          workflow: 'Getting Corona Test Date',
-          step: `launching Corona Test Date request`,
-          user: userId,
-          investigation: investigationId
-        });
+        getCoronaTestDateLogger.info(`launching Corona Test Date request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx:113]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx [hunk 11]
```diff
@@ -165,29 +130,19 @@ const ExposuresAndFlights: React.FC<Props> = ({ id }: Props): JSX.Element => {
...
-        wasInEilat: result?.data?.wasInEilat,
-        wasInDeadSea: result?.data?.wasInDeadSea
-      }
-      return resortData;
+    const fetchResortsDataLogger = logger.setup({
+      workflow: 'Fetching investigated patient resorts data',
+      investigation: investigationId,
+      user: userId
+    });
+    fetchResortsDataLogger.info(`launching investigated patient resorts request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx:138]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx [hunk 12]
```diff
@@ -165,29 +130,19 @@ const ExposuresAndFlights: React.FC<Props> = ({ id }: Props): JSX.Element => {
...
+    const result = await axios.get('investigationInfo/resorts/' + investigatedPatientId);
+    fetchResortsDataLogger.info(`got investigated patient resorts response successfully`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx:140]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 13]
```diff
@@ -40,105 +40,61 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
...
-            severity: Severity.LOW,
+        const getCoronaTestDateLogger = logger.setup({
             workflow: 'Getting Corona Test Date',
-            step: `launching Corona Test Date request`,
-            user: userId,
-            investigation: epidemiologyNumber
+            investigation: epidemiologyNumber,
+            user: userId
         });
+        getCoronaTestDateLogger.info(`launching Corona Test Date request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:48]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 14]
```diff
@@ -40,105 +40,61 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
...
-            severity: Severity.LOW,
+        const loadInteractionsLogger = logger.setup({
             workflow: 'Fetching Interactions',
-            step: `launching interactions request`,
-            user: userId,
-            investigation: epidemiologyNumber
+            investigation: epidemiologyNumber,
+            user: userId
         });
+        loadInteractionsLogger.info(`launching interactions request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:86]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 15]
```diff
@@ -225,35 +186,14 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
...
             if (result.value) {
-                logger.info({
-                    service: Service.CLIENT,
-                    severity: Severity.LOW,
-                    workflow: 'Deleting Interaction',
-                    step: `launching interaction delete request`,
-                    user: userId,
-                    investigation: epidemiologyNumber
-                });
+                deletingInteractionsLogger.info(`launching interaction delete request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:189]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 16]
```diff
@@ -225,35 +186,14 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
...
                 }).then(() => {
-                    logger.info({
-                        service: Service.CLIENT,
-                        severity: Severity.LOW,
-                        workflow: 'Deleting Interaction',
-                        step: `interaction was deleted successfully`,
-                        user: userId,
-                        investigation: epidemiologyNumber
-                    });
+                    deletingInteractionsLogger.info(`interaction was deleted successfully`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:193]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 17]
```diff
@@ -276,35 +221,14 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
...
             if (result.value) {
-                logger.info({
-                    service: Service.CLIENT,
-                    severity: Severity.LOW,
-                    workflow: 'Deleting Contacted Person',
-                    step: `launching interaction delete request`,
-                    user: userId,
-                    investigation: epidemiologyNumber
-                });
+                deleteContactedPersonLogger.info(`launching interaction delete request`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:224]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts [hunk 18]
```diff
@@ -276,35 +221,14 @@ const useInteractionsTab = (parameters: useInteractionsTabParameters): useIntera
...
                 }).then(() => {
-                    logger.info({
-                        service: Service.CLIENT,
-                        severity: Severity.LOW,
-                        workflow: 'Deleting Contacted Person',
-                        step: `interaction was deleted successfully`,
-                        user: userId,
-                        investigation: epidemiologyNumber
-                    });
+                    deleteContactedPersonLogger.info(`interaction was deleted successfully`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/useInteractionsTab.ts:228]: singel quote

File: client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/usePersonalInfoTab.ts [hunk 19]
```diff
@@ -180,62 +130,37 @@ const usePersonalInfoTab = (parameters: usePersoanlInfoTabParameters): usePerson
...
             if (res && res.data && res.data.data) {
-                logger.info({
-                    service: Service.CLIENT,
-                    severity: Severity.LOW,
-                    workflow: 'Fetching Education Sub Occupation by City',
-                    step: `got results from the server`,
-                    user: userId,
-                    investigation: epidemiologyNumber
-                });
+                educationSubOccupationsLogger.info(`got results from the server`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/usePersonalInfoTab.ts:163]: singel quote

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 20]
```diff
@@ -234,13 +200,11 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
-        logger.info({
-            service: Service.CLIENT,
-            severity: Severity.LOW,
+        const countyUsersLogger = logger.setup({
             workflow: 'Getting group users',
-            step: `requesting the server the connected admin's group users`,
             user: user.id
-        })
+        });
+        countyUsersLogger.info(`requesting the server the connected admin's group users`,Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:207]: singel quote

File: client/src/Utils/ControllerHooks/useExposuresSaving.ts [hunk 21]
```diff
@@ -26,15 +26,12 @@ const useExposuresSaving = (exposuresAndFlightsVariables: ExposureAndFlightsDeta
...
-            severity: Severity.LOW,
+        const saveResortsDataLogger = logger.setup({
             workflow: 'Saving investigated patient resort data',
-            step: 'launching the server request',
-            investigation: epidemiologyNumber,
-            user: userId
+            user: userId,
+            investigation: epidemiologyNumber
         });
+        saveResortsDataLogger.info('launching the server request',Severity.LOW)
```
Comment (ItayBenMoshe) [inline client/src/Utils/ControllerHooks/useExposuresSaving.ts:34]: add space after , (same in others)

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

File: client/src/logger/logger.ts [hunk 22]
```diff
@@ -32,7 +32,6 @@ class Logger {
 
     _buildLogMessage(partialLogMessage: MethodsLogMessage, logType: LogType) {
         const logMessage: LogMessage = {
-            service: Service.CLIENT,
```
Comment (NaamanTsur) [inline client/src/logger/logger.ts:35]: We still need to define the client service for the logs.

**Review by NaamanTsur — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 23]
```diff
@@ -530,12 +530,12 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
                     updatedCounty: newSelectedCounty.id,
                 }).then(() => timeout(1900).then(() => {
-                    changeCountyLogger.info('changed the county successfully',  Severity.LOW);
+                    changeCountyLogger.info('changed the county successfully', Severity.LOW);
                     setSelectedRow(UNDEFINED_ROW);
                     setRows(rows.filter((row: InvestigationTableRow) => row.epidemiologyNumber !== indexedRow.epidemiologyNumber));
                 }))
                 .catch((error) => {
-                    changeCountyLogger.error('couldnt change the county due to ' + error,  Severity.LOW);
+                    changeCountyLogger.error('couldnt change the county due to ' + error, Severity.LOW);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:538]: $ syntax

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 24]
```diff
@@ -586,7 +586,7 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
                         setSelectedRow(UNDEFINED_ROW);
                     }))
                     .catch((error) => {
-                        changeDeskLogger.error('couldnt chang the desk due to ' + error,  Severity.HIGH);
+                        changeDeskLogger.error('couldnt chang the desk due to ' + error, Severity.HIGH);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:589]: same

**Review by ItayBenMoshe — APPROVED** [review]

**Review by NaamanTsur — APPROVED** [review]

**Pull request merged**